Makefile

```diff
@@ -68,7 +40,8 @@ ifeq ($(UNAME_S),Linux)
 	DUCKDB_LIB = libduckdb.so
 endif
 
-all: duckdb $(OBJS) .depend
+all: $(OBJS) .depend
+all-lib: duckdb
```

Comment:
I'm not sure I understand the point of this change.

Reply:
This was the most effective way I could find to make sure duckdb was built before pg_duckdb starts building.

I'm open to other suggestions, I'm certainly not a `make` wizard. 😁

Reply:
ok this isnt having the effect I thought it was having.

I also tried:

```
$(OBJS): duckdb
```

but this caused all of pg_duckdb to be rebuilt on every `make`, which didn't make sense to me.

Reply:
I think this might work:
```
$(OBJS): $(FULL_DUCKDB_LIB)
```

Things depending on a .PHONY means they always needs to be rebuilt (because make cannot do timestamp checks on a .PHONY target)

Reply:
If that doesn't work let's remove this all/all-lib change from this PR, because all the other things look reasonable.

Reply:
Finally realized this was some kind of race condition with the `install` step, so this was inadvertently exposed by using `make install` rather than `make` then `make install`. Adding a dependency to completely finish the build before installing seems to have fixed it.